ercc/Makefile

```diff
@@ -32,6 +32,16 @@ else
 SGX_MODE_DEPENDENCY=
 endif
 
+ias-keys:
```

Comment:
what if I tested HW, then SIM, then HW?
This overwrites my keys and I need to start over with the configuration but... only I after I find out that it does not work!!!

Reply:
no, it does not override, it creates files only if they do not exist (at least that's the intention)

Reply:
point to you! :P

New round: if I test in SIM then HW, files already exist but...they are wrong.
So, should we separate these credentials?

Reply:
another solution is in ercc/makefile: create .sgx_mode file and put a dependency
or maybe we could graduate that file to the top folder

Reply:
different files requires different fabric configs which i don't think is an appealing solution, at least not better than this one. Your case is valid but only for somebody who hasn't followed the instruction. 

iff we get access to SGX_MODE in ercc/attestation/ias_credentials/decoration.go we also could just do it there but not easy via go build parameters (and i don't really like the templating stuff to extend it here as well). I was thinking first on whether could do it based on the attestation type (which for sim always should be mock?) but somehow didn't see exactly how i access it in decoration.go.

So for now i still will leave it as-is ..

Reply:
> Your case is valid but only for somebody who hasn't followed the instruction.
I disagree. 
As a user, _I_ should follow the instruction, and _the framework_ should warn me properly when things are not set up properly.
Here there are four problems:
1) the connection to IAS will fail - and the user has to figure out that this is due to the credentials
2) the mock SIM credentials, created by the framework, are accepted by the framework in HW mode.
3) this has already happened in PDO!

> iff we get access to SGX_MODE in ercc/attestation/ias_credentials/decoration.go we also could just do it there but not easy via go build parameters (and i don't really like the templating stuff to extend it here as well). I was thinking first on whether could do it based on the attestation type (which for sim always should be mock?) but somehow didn't see exactly how i access it in decoration.go.

This is complicated!

Reply:
well, i'm not arguing this PR is a perfect solution but it is (still) the best simple fix i know of to address the issue identified by Ann.  It is certainly better than the status-quo where the documentation was misleading  :-)

The related code almost certainly also will change quite a bit with new ercc, proper chaincode where the decorators should go away and the values are directly accessed from system cc and alike, so we are talking anyway only of an intermediary fix.